src/EditorFeatures/Core/SymbolSearch/RoslynSymbolResult.cs

```diff
@@ -1,22 +1,73 @@
 
-        public RoslynSymbolResult(DefinitionItem definition)
+        public RoslynSymbolResult(RoslynSymbolSource source, DefinitionItem definition)
         {
+            this.source = source;
             this.definition = definition;
+            Owner = source;
+            Name = definition.ToString();
+            Origin = GetLocalOrigin();
+            ClassifiedDefinition = new ClassifiedTextElement(definition.NameDisplayParts.Select(n => new ClassifiedTextRun(n.Style.ToString(), n.Text)));
+            ClassifiedContext = new ClassifiedTextElement(definition.DisplayParts.Select(n => new ClassifiedTextRun(n.Style.ToString(), n.Text)));
+            var sampleLocation = definition.SourceSpans.First(); // TODO: handle all locations
+            PersistentSpan = source.ServiceProvider.PersistentSpanFactory.Create(sampleLocation.Document.FilePath, AsSpan(sampleLocation.SourceSpan), SpanTrackingMode.EdgeInclusive);
         }
 
-        public IPersistentSpan PersistentSpan => throw new System.NotImplementedException();
+        public RoslynSymbolResult(RoslynSymbolSource source, SourceReferenceItem reference)
+            : this(source, reference.Definition)
+        {
+            this.reference = reference;
+            Name = reference.ToString(); // which one runs first?
+        }
+
+        private Span AsSpan(TextSpan sourceSpan)
+        {
+            return new Span(sourceSpan.Start, sourceSpan.Length);
+        }
+
+        private ISymbolOriginDefinition GetLocalOrigin()
+        {
+            if (LocalSymbolOrigin == null)
+            {
+                source.ServiceProvider.SymbolSearchBroker.SymbolOriginDefinitions.TryGetValue(PredefinedSymbolOrigins.LocalCode, out LocalSymbolOrigin);
+            }
+            return LocalSymbolOrigin;
+        }
 
-        public override ISymbolOriginDefinition Origin { get => throw new System.NotImplementedException(); protected set => throw new System.NotImplementedException(); }
-        public override ISymbolSource Owner { get => throw new System.NotImplementedException(); protected set => throw new System.NotImplementedException(); }
-        public override string Name { get => throw new System.NotImplementedException(); protected set => throw new System.NotImplementedException(); }
+        public IPersistentSpan PersistentSpan { get; private set; }
+
+        public override ISymbolOriginDefinition Origin { get; protected set; }
+        public override ISymbolSource Owner { get; protected set; }
+        public override string Name { get; protected set; }
+
+        public ClassifiedTextElement ClassifiedDefinition { get; private set; }
+
+        public ClassifiedTextElement ClassifiedContext { get; private set; }
+
+        public Span HighlightSpan => throw new System.NotImplementedException();
+
+        public Task NavigateToAsync(CancellationToken token)
+        {
+            // how to get the workspace?
```

Comment:
you have the Document with the definition.  You can get the workspace off of that.

Reply:
however, you shouldn't nav to documents (because it may not be a reference backed by a document).  The FindRefs system should show you how to navigate to any result.  This will then work for things like metadata, or xaml, etc.